### PR TITLE
made navbar responsive in mobile and added border animation on navbar sections

### LINKED ISSUE #859: Nav Bar Unresponsiveness Bug in Mobile Screens [React]
## Current Navbar/Sidebar in Mobile Screens

Navbar_current

Navbar_current_1

## React Version
Navbar_unresponsive

- There is a micro animation on clicking the hamburger button [ [Reference](https://codepen.io/RRoberts/pen/ZBYaJr) 2nd row, 2nd column is a preference)
- A drop down effect spanning the entire page 
- An illustration at the bottom (Feel free to change this)

### PR DESCRIPTION
**Pull request by GiotisK**
## Corresponding Issues:
fixes #859

## PR Summary:
- Added hambuger button (as component) and toggle animation (as referenced in the issue)
- Added dropdown animation using the AnimatedHeight library (height animations with pure css are difficult to implement)
- Cleaned up the Navbar.js  component code
- Added color variables

## Outcome Video:  

https://user-images.githubusercontent.com/10963096/126037024-49480c4b-2d22-4b62-90d2-da4bbed0738c.mp4

### CODE REVIEW

**Comment by GiotisK** [comment]
Not sure what caused the failed deployment. Can you check it and let me know so I can fix it? Thank you

**Review by anushbhatia — CHANGES_REQUESTED** [review]
Add the hamburger files in the components folder and pull it in navbar

File: src/components/Navbar/HambugerButton.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import './HamburgerButton.css'
+
+const HamburgerButton = (props) => {
+    return(
+        <div onClick = {props.onClick} className={`hamburger ${props.activated ? 'is-active' : ''}`} id="hamburger-6">
```
Comment (anushbhatia) [inline src/components/Navbar/HambugerButton.js:6]: why id= hamburger-6 can't it be hamburger

**Review by anushbhatia — COMMENTED** [review]
Great Work with the comments 😎

**Comment by GiotisK** [comment]
@anushbhatia I've done the changes you proposed ! 🚀

**Review by smaranjitghose — APPROVED** [review]

**Pull request merged**